cmd/kube-controller-manager/app/core.go

```diff
@@ -141,7 +150,7 @@ func startNodeLifecycleController(ctx ControllerContext) (http.Handler, bool, er
 		ctx.InformerFactory.Core().V1().Nodes(),
 		//ctx.InformerFactory.Apps().V1().DaemonSets(),
 		// node lifecycle controller uses existing cluster role from node-controller
-		ctx.ClientBuilder.ClientOrDie("node-controller"),
+		kubeclient,
```

Comment:
this should not  be change to kubeclient. those are original controller clientset we should keep as it. in addition to that, the kubeclient can be empty with the code above ( if - else block )

Reply:
That is a good suggestion. I will update it.